### PR TITLE
Patch for Issue #257

### PR DESCRIPTION
**Pull request by ArnabRaxit**

### CODE REVIEW

File: core/src/main/java/hivemall/evaluation/R2UDAF.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        }
+
+        double getR2() {
+        	double avg_actuals = this.sum_actuals/this.count;
+        	double total_sum_of_squares = 0d;
+        	
+        	for (Double a : actuals) {
+        		total_sum_of_squares += Math.pow(a-avg_actuals, 2.d);
+			}
+            return 1 - this.residual_sum_of_squares/total_sum_of_squares;
```
Comment (myui) [inline core/src/main/java/hivemall/evaluation/R2UDAF.java]: Please avoid divide by zero exception where total_sum_of_squares is 0.

File: core/src/main/java/hivemall/evaluation/R2UDAF.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+        void merge(PartialResult other) {
+            residual_sum_of_squares += other.residual_sum_of_squares;
+            this.actuals.addAll(other.actuals);
+            this.sum_actuals += other.sum_actuals;
+            count += other.count;
+        }
+
+        double getR2() {
+        	double avg_actuals = this.sum_actuals/this.count;
```
Comment (myui) [inline core/src/main/java/hivemall/evaluation/R2UDAF.java]: Please avoid divide by zero where count is 0 and return 1.

**Comment by myui** [comment]
@ArnabRaxit Thank you for the contribution. Review comments are attached. Please revise about it.

**Comment by myui** [comment]
Oops... When `partial != null`, count will be `counter >= 1`.

So, just avoiding divide by zero exception and return 1.d where total_sum_of_squares is 0.d is enough.

**Comment by myui** [comment]
I handled the above issue. Merged. Thanks!

**Comment by ArnabRaxit** [comment]
Thanks for taking care of it :)

**Comment by myui** [comment]
Hi @ArnabRaxit 

I sent a e-mail to you. Did you get it? (I'm not sure your e-mail address).

Thanks,
Makoto

**Comment by ArnabRaxit** [comment]
Yes I did. I shall respond by tomorrow. Thanks :)

**Comment by myui** [comment]
@ArnabRaxit Thanks :-)

**Pull request merged**